File: source java/organism/plants/SosnowskysHogweed.java
package organism.plants;

import java.awt.Color;
import organism.CollisionResult;
import organism.Organism;
import organism.animals.CyberSheep;
import organism.manager.Organisms;
import utilities.position.Position;
import utilities.savingsystem.Loader;
import virtualworld.World;

public class SosnowskysHogweed extends Plant
{
    private static final int STRENGTH = 10;

    public SosnowskysHogweed(Position position, World world) 
    {
        super(STRENGTH, position, world);
    }
    
    public SosnowskysHogweed(Loader loader, World world, Position position)
    {
        super(loader, world, position);
    }
    
    @Override
    public void Act()
    {
        Position[] neighbours = GetPosition().GetNeighbours();
        for(Position position : neighbours)
        {
            Organism neighbourOrganism = GetWorld().GetOrganism(position);
            if(neighbourOrganism != null && !(neighbourOrganism instanceof CyberSheep) && !(neighbourOrganism instanceof Plant))
            {
                GetWorld().AddToLogs(neighbourOrganism.toString() + " got urticated.");
                neighbourOrganism.Die();
            }
        }
    }
    
    @Override
    public CollisionResult Collide(Organism organism)
    {
        if(!(organism instanceof CyberSheep))
            organism.Die();
        return CollisionResult.PLANT;
    }
    
    @Override 
    public Organisms GetEnum()
    {
        return Organisms.SOSNOWSKYS_HOGWEED;
    }
    
    @Override
    public Color GetColor()
    {
        return Color.magenta;
    }
    
    @Override
    public String toString()
    {
        return "Sosnowsky's Hogweed";
    }
}

File: source python/Organisms/Plants/SosnowskysHogweed.py
from Application.Organisms.Plants.Plant import Plant
from Application.Organisms.Organism import ColisionResult
import Application.Organisms.Animals.CyberSheep
from Application.Utilities import Colors


class SosnowskysHogweed(Plant):

    STRENGTH = 10

    def __init__(self, position, world, **kwargs):
        super().__init__(SosnowskysHogweed.STRENGTH, position, world, **kwargs)

    def Collide(self, organism):
        if not isinstance(organism, Application.Organisms.Animals.CyberSheep.CyberSheep):
            print(str(organism) + " dies after contact with " + str(self))
            organism.Die()
        return ColisionResult.PLANT

    def Act(self):
        neighbours = self.GetPosition().GetAllNeighbours()
        for neighbour in neighbours:
            organism = self.GetWorld().GetOrganism(neighbour)
            if organism is not None and not isinstance(organism, Application.Organisms.Animals.CyberSheep.CyberSheep) and (not isinstance(organism, Plant)):
                organism.Die()
                print(str(organism) + " dies after contact with " + str(self))
        self._Spread()

    @staticmethod
    def __str__():
        return "Sosnowsky's hogweed"

    def GetColor(self):
        return Colors.MAGENTA

    def GetLetter(self):
        return "B"